src/test/storage/adaptive_radix_tree_index_test.cpp

```diff
@@ -140,7 +176,62 @@ TEST_F(AdaptiveRadixTreeIndexTest, VectorOfRandomInts) {
       expected_lower += 2;
     }
   }
-  EXPECT_EQ(index->upper_bound({99999}), index->cend());
+
+  for (int search_item = 0; search_item < static_cast<int>(3 * test_size); ++search_item) {
+    if (search_item % 2 == 0 && search_item < static_cast<int>(2 * test_size)) {
+      // all multiples of two within range of $ints should exist
+      EXPECT_NE(index->lower_bound({search_item}), index->cend());
+      continue;
+    }
+
+    // search for elements not existing
+    EXPECT_EQ(index->lower_bound({search_item}), index->upper_bound({search_item}));
+  }
+}
+
+TEST_F(AdaptiveRadixTreeIndexTest, SimpleTest) {
+  std::vector<int> values = {0, 0, 0, 0, 0, 17, 17, 17, 99, std::numeric_limits<int>::max()};
+
+  auto segment = create_dict_segment_by_type<int>(DataType::Int, values);
+  auto index = std::make_shared<AdaptiveRadixTreeIndex>(std::vector<std::shared_ptr<const BaseSegment>>({segment}));
+
+  EXPECT_EQ(*index->cbegin(), 0);
+  EXPECT_EQ(*index->lower_bound({0}), 0);
+  EXPECT_EQ(*index->upper_bound({0}), 5);
+  EXPECT_EQ(*index->lower_bound({17}), 5);
+  EXPECT_EQ(*index->upper_bound({17}), 8);
+  EXPECT_EQ(*index->lower_bound({99}), 8);
+  EXPECT_EQ(*index->upper_bound({99}), 9);
+  EXPECT_EQ(*index->lower_bound({std::numeric_limits<int>::max()}), 9);
+  EXPECT_EQ(index->upper_bound({std::numeric_limits<int>::max()}), index->cend());
+}
+
+/**
+* The following two tests try to test two rather extreme tests that both test the node overflow
```

Comment:
Lots of `test` in that line, maybe `cases`?

Reply:
Done.